File: apps/omnimereduapp/lib/domain/usecases/school/get_schools_by_level_usecase.dart
// domain/usecases/school/get_schools_by_level_usecase.dart
import '../../../core/constants/enum_constant.dart';
import '../../entities/school/school_search_entity.dart';
import '../../repositories/school/school_repository.dart';

class GetSchoolsByLevelUseCase {
  final SchoolRepository repository;

  GetSchoolsByLevelUseCase(this.repository);

  Future<List<SchoolSearchEntity>> call(
    EducationSystemLevelsEnum educationLevel,
  ) async {
    return await repository.getSchoolsByLevel(educationLevel);
  }
}

File: services/user_service/src/domain/usecases/school/GetSchoolsByLevelUseCase.ts
import { ISchoolRepository } from "../../repositories/ISchoolRepository";

export class GetSchoolsByLevelUseCase {
  constructor(private schoolRepository: ISchoolRepository) {}

  async execute(params: {
    educationLevel: string;
    search?: string;
  }): Promise<any[]> {
    return await this.schoolRepository.getSchoolsByLevel(params);
  }
}
